slicingTool/runProjectSlicingTool.sh

```diff
@@ -26,14 +34,20 @@ if [[ -n $DB && -n $USER && -n $PASS ]]
 then
     if [[ $USER =~ "curator" ]]
     then
-	echo Dropping $DB ...
-	mysql -u$USER -p$PASS -e "drop database if exists $DB"
-#	mysql -u$USER -p$PASS -e "drop database if exists ${DB}_myisam"
-#	mysql -u$USER -p$PASS -e "create database ${DB}_myisam"
-#	mysql -u$USER -p$PASS -e "drop database if exists gk_central"
-#	mysql -u$USER -p$PASS -e "create database gk_central"
+        # Attempts to find and use database $DB and if successful it is backed up
+        # before being dropped
+        if mysql -u $USER -p$PASS -e "use $DB";
+        then
+            echo Backing up $DB ...
+            mysqldump -u$USER -p$PASS $DB > $DB.dump
+        else
+            echo Database $DB does not exist - no need to create back up
```

Comment:
I'm not sure that message is very accurate - what if the user got the username/password wrong? The database might exist, but it will not be backed up if the username/password cause the earlier check to fail. I think an error here should be a hard stop in this process. I'm not sure the process should be allowed to continue if the backup failed.

Reply:
Good point.  I think I will need to capture and parse the error.  It will also fail if the database simply doesn't exist.  If that's the case, the process should continue.  You're right that if the user and password fail, it should stop.